### PR TITLE
Brave news android rss

### PR DESCRIPTION
**Pull request by alexsafe**
Resolves https://github.com/brave/brave-browser/issues/21272

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/settings/BraveNewsPreferences.java [hunk 1]
```diff
@@ -44,10 +52,14 @@
     public static final String PREF_SHOW_OPTIN = "show_optin";
     public static final String PREF_SOURCES_SECTION = "your_sources_section";
     public static final String PREF_ADD_SOURCES = "add_source_news";
+    public static final String PREF_ADD_RSS_SOURCES = "news_source_1";
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/settings/BraveNewsPreferences.java:55]: It seems this constant does not used

File: android/java/org/chromium/chrome/browser/settings/BraveNewsPreferences.java [hunk 2]
```diff
@@ -44,10 +52,14 @@
...
+    private static final String PREF_RSS_SOURCES = "rss_sources";
 
     private ChromeSwitchPreference mTurnOnNews;
     private ChromeSwitchPreference mShowNews;
+    private EditTextPreference addSource;
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/settings/BraveNewsPreferences.java:60]: nit: naming for the privateMember

**Review by SergeyZhukovsky — APPROVED** [review]
++

File: android/java/org/chromium/chrome/browser/settings/BraveNewsPreferences.java [hunk 3]
```diff
@@ -74,20 +89,69 @@ public void onCreatePreferences(@Nullable Bundle savedInstanceState, String root
...
             mCategsPublishers.put("All Sources", allPublishers);
             addCategs(mCategsPublishers);
+            addRss(rssPublishers);
         });
     }
 
+    private void addRss(List<Publisher> publishers) {
+        for (Publisher publisher : publishers) {
+            if (publisher.type != PublisherType.DIRECT_SOURCE) {
+                return;
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/settings/BraveNewsPreferences.java:111]: as per https://github.com/brave/brave-core/pull/12379/files#diff-2ddfa5e09ee57aab48d02c29d3a82bc7f71f4c1e3eeb94559e83e06c047a4d45R99
`publishers` contains only those which are `publisher.type == PublisherType.DIRECT_SOURCE`
so this `return` statement will not be executed ever.

What is the intention of this check?
Maybe it is enough to put `assert(publisher.type == PublisherType.DIRECT_SOURCE)`?

File: android/java/res/layout/pref_edittext_dialog.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--* Copyright (c) 2021 The Brave Authors. All rights reserved.
```
Comment (AlexeyBarabash) [inline android/java/res/layout/pref_edittext_dialog.xml:2]: nit: year  2022

File: android/java/res/xml/brave_news_preferences.xml [hunk 5]
```diff
@@ -19,11 +19,16 @@
         android:summaryOn="@string/brave_news_optin_subtitle"
         android:summaryOff="@string/brave_news_optin_subtitle" />
 
-<!--     <PreferenceCategory
+    <PreferenceCategory
```
Comment (AlexeyBarabash) [inline android/java/res/xml/brave_news_preferences.xml:22]: what was that commented out element :+1:

**Review by AlexeyBarabash — APPROVED** [review]
looks good :+1: , I have only several minor notices

File: android/java/res/xml/brave_news_preferences.xml (hunk 5 above)
Comment (alexsafe) [reply android/java/res/xml/brave_news_preferences.xml:22]: @AlexeyBarabash i think it was an old formatting/commit thing, nothing important should be missing

File: android/java/org/chromium/chrome/browser/settings/BraveNewsPreferences.java (hunk 3 above)
Comment (alexsafe) [reply android/java/org/chromium/chrome/browser/settings/BraveNewsPreferences.java:111]: @AlexeyBarabash yeah that was exactly the point. a double check that only those sources are added to the RSS section

**Comment by AlexeyBarabash** [comment]
thanks  for resolving all the notices @alexsafe

File: android/java/org/chromium/chrome/browser/settings/BraveNewsPreferences.java (hunk 1 above)
Comment (alexsafe) [reply android/java/org/chromium/chrome/browser/settings/BraveNewsPreferences.java]: > thanks for resolving all the notices @alexsafe

thanks for pointing them @AlexeyBarabash

**Pull request merged**